src/node-utils.ts

```diff
@@ -30,72 +32,73 @@ const ASSIGNMENT_OPERATORS = [
-TOKEN_TO_TEXT[SyntaxKind.CloseBracketToken] = ']';
-TOKEN_TO_TEXT[SyntaxKind.DotToken] = '.';
-TOKEN_TO_TEXT[SyntaxKind.DotDotDotToken] = '...';
-TOKEN_TO_TEXT[SyntaxKind.SemicolonToken] = ';';
-TOKEN_TO_TEXT[SyntaxKind.CommaToken] = ',';
-TOKEN_TO_TEXT[SyntaxKind.LessThanToken] = '<';
-TOKEN_TO_TEXT[SyntaxKind.GreaterThanToken] = '>';
-TOKEN_TO_TEXT[SyntaxKind.LessThanEqualsToken] = '<=';
-TOKEN_TO_TEXT[SyntaxKind.GreaterThanEqualsToken] = '>=';
-TOKEN_TO_TEXT[SyntaxKind.EqualsEqualsToken] = '==';
-TOKEN_TO_TEXT[SyntaxKind.ExclamationEqualsToken] = '!=';
-TOKEN_TO_TEXT[SyntaxKind.EqualsEqualsEqualsToken] = '===';
-TOKEN_TO_TEXT[SyntaxKind.InstanceOfKeyword] = 'instanceof';
-TOKEN_TO_TEXT[SyntaxKind.ExclamationEqualsEqualsToken] = '!==';
-TOKEN_TO_TEXT[SyntaxKind.EqualsGreaterThanToken] = '=>';
-TOKEN_TO_TEXT[SyntaxKind.PlusToken] = '+';
-TOKEN_TO_TEXT[SyntaxKind.MinusToken] = '-';
-TOKEN_TO_TEXT[SyntaxKind.AsteriskToken] = '*';
-TOKEN_TO_TEXT[SyntaxKind.AsteriskAsteriskToken] = '**';
-TOKEN_TO_TEXT[SyntaxKind.SlashToken] = '/';
-TOKEN_TO_TEXT[SyntaxKind.PercentToken] = '%';
-TOKEN_TO_TEXT[SyntaxKind.PlusPlusToken] = '++';
-TOKEN_TO_TEXT[SyntaxKind.MinusMinusToken] = '--';
-TOKEN_TO_TEXT[SyntaxKind.LessThanLessThanToken] = '<<';
-TOKEN_TO_TEXT[SyntaxKind.LessThanSlashToken] = '</';
-TOKEN_TO_TEXT[SyntaxKind.GreaterThanGreaterThanToken] = '>>';
-TOKEN_TO_TEXT[SyntaxKind.GreaterThanGreaterThanGreaterThanToken] = '>>>';
-TOKEN_TO_TEXT[SyntaxKind.AmpersandToken] = '&';
-TOKEN_TO_TEXT[SyntaxKind.BarToken] = '|';
-TOKEN_TO_TEXT[SyntaxKind.CaretToken] = '^';
-TOKEN_TO_TEXT[SyntaxKind.ExclamationToken] = '!';
-TOKEN_TO_TEXT[SyntaxKind.TildeToken] = '~';
-TOKEN_TO_TEXT[SyntaxKind.AmpersandAmpersandToken] = '&&';
-TOKEN_TO_TEXT[SyntaxKind.BarBarToken] = '||';
-TOKEN_TO_TEXT[SyntaxKind.QuestionToken] = '?';
-TOKEN_TO_TEXT[SyntaxKind.ColonToken] = ':';
-TOKEN_TO_TEXT[SyntaxKind.EqualsToken] = '=';
-TOKEN_TO_TEXT[SyntaxKind.PlusEqualsToken] = '+=';
-TOKEN_TO_TEXT[SyntaxKind.MinusEqualsToken] = '-=';
-TOKEN_TO_TEXT[SyntaxKind.AsteriskEqualsToken] = '*=';
-TOKEN_TO_TEXT[SyntaxKind.AsteriskAsteriskEqualsToken] = '**=';
-TOKEN_TO_TEXT[SyntaxKind.SlashEqualsToken] = '/=';
-TOKEN_TO_TEXT[SyntaxKind.PercentEqualsToken] = '%=';
-TOKEN_TO_TEXT[SyntaxKind.LessThanLessThanEqualsToken] = '<<=';
-TOKEN_TO_TEXT[SyntaxKind.GreaterThanGreaterThanEqualsToken] = '>>=';
-TOKEN_TO_TEXT[SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken] = '>>>=';
-TOKEN_TO_TEXT[SyntaxKind.AmpersandEqualsToken] = '&=';
-TOKEN_TO_TEXT[SyntaxKind.BarEqualsToken] = '|=';
-TOKEN_TO_TEXT[SyntaxKind.CaretEqualsToken] = '^=';
-TOKEN_TO_TEXT[SyntaxKind.AtToken] = '@';
-TOKEN_TO_TEXT[SyntaxKind.InKeyword] = 'in';
-TOKEN_TO_TEXT[SyntaxKind.UniqueKeyword] = 'unique';
-TOKEN_TO_TEXT[SyntaxKind.KeyOfKeyword] = 'keyof';
-TOKEN_TO_TEXT[SyntaxKind.NewKeyword] = 'new';
-TOKEN_TO_TEXT[SyntaxKind.ImportKeyword] = 'import';
+const TOKEN_TO_TEXT: { [k: number]: string } = {
```

Comment:
Could this be `SyntaxKind` instead?

```suggestion
const TOKEN_TO_TEXT: { [kind: SyntaxKind]: string } = {
```

Reply:
sadly no, Typescript is not allowing only string and number as key

Reply:
https://github.com/Microsoft/TypeScript/issues/2491

Reply:
hmm, they added mappedTypes hmm https://github.com/Microsoft/TypeScript/pull/12114

Reply:
@j-f1 
what do you think about:
```
{
  readonly [P in keyof typeof SyntaxKind]: string;
}
```
but i will have to define all of them

Reply:
ok i updated it to:
```ts
const TOKEN_TO_TEXT: { readonly [P in keyof typeof SyntaxKind]?: string } = {
```